internal/saptune/saptune_test.go

```diff
@@ -0,0 +1,232 @@
+package saptune_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/suite"
+	"github.com/trento-project/workbench/internal/saptune"
+	"github.com/trento-project/workbench/internal/support/mocks"
+	"github.com/trento-project/workbench/test/helpers"
+)
+
+type SaptuneClientTestSuite struct {
+	suite.Suite
+	mockExecutor *mocks.MockCmdExecutor
+	logger       *logrus.Entry
+}
+
+func TestSaptuneClient(t *testing.T) {
+	suite.Run(t, new(SaptuneClientTestSuite))
+}
+
+func (suite *SaptuneClientTestSuite) SetupTest() {
+	suite.mockExecutor = mocks.NewMockCmdExecutor(suite.T())
+	suite.logger = logrus.NewEntry(logrus.StandardLogger())
+}
+
+func (suite *SaptuneClientTestSuite) TestVersionCheckFailureBecauseUnableToDetectVersion() {
+	ctx := context.Background()
+
+	suite.mockExecutor.On(
+		"Exec",
+		saptuneVersionRetrieverCommandArguments(ctx)...,
```

Comment:
super nitpick:
personally, I find the function usage worst than having duplicated code. They require a lot of scrolling, and reduce the visibility of what we really are testing. 
Anyway, just personal preference, I don't think some duplication is tests is bad

Reply:
I don't mind eitherway. If you prefer the other way I am happy to do so. Changed in the last commit